src/foundry/foundry.js/canvas.d.ts

```diff
@@ -17,401 +20,498 @@
+
+    foreground: ForegroundLayer | undefined;
+
+    sounds: SoundsLayer | undefined;
+
+    lighting: LightingLayer | undefined;
+
+    sight: SightLayer | undefined;
+
+    effects: EffectsLayer | undefined;
+
+    controls: ControlsLayer | undefined;
+
+    /**
+     * When re-drawing the canvas, first tear down or discontinue some existing processes
+     */
+    tearDown(): Promise<void>;
+
+    /**
+     * Draw the game canvas.
+     * @param scene - A specific Scene document to render on the Canvas
+     * @returns A Promise which resolves once the Canvas is fully drawn
+     */
+    draw(scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>): Promise<this>;
+
+    /**
+     * Get the canvas active dimensions based on the size of the scene's map.
+     * We expand the image size by a factor of 1.5 and round to the nearest 2x grid size.
+     * The rounding accomplishes that the padding buffer around the map always contains whole grid spaces.
+     * @see {@link documents.BaseScene.getDimensions}
+     * @param data - The scene dimensions data being established
+     */
+    static getDimensions(data: Canvas.DimensionsData): Canvas.Dimensions;
+
+    /**
+     * Once the canvas is drawn, initialize control, visibility, and audio states
+     */
+    protected _initialize(): Promise<void>;
+
+    /**
+     * Initialize the starting view of the canvas stage
+     * If we are re-drawing a scene which was previously rendered, restore the prior view position
+     * Otherwise set the view to the top-left corner of the scene at standard scale
+     */
+    protected _initializeCanvasPosition(): void;
+
+    /**
+     * Initialize a CanvasLayer in the activation state
+     */
+    protected _initializeCanvasLayer(): void;
+
+    /**
+     * Initialize a token or set of tokens which should be controlled.
+     * Restore controlled and targeted tokens from before the re-draw.
+     */
+    protected _initializeTokenControl(): void;
+
+    /**
+     * Get a reference to the a specific CanvasLayer by it's name
+     * @param layerName - The name of the canvas layer to get
+     */
+    getLayer(layerName: 'BackgroundLayer'): BackgroundLayer | null;
+    getLayer(layerName: 'DrawingsLayer'): DrawingsLayer | null;
+    getLayer(layerName: 'GridLayer'): GridLayer | null;
+    getLayer(layerName: 'WallsLayer'): WallsLayer | null;
+    getLayer(layerName: 'TemplateLayer'): TemplateLayer | null;
+    getLayer(layerName: 'NotesLayer'): NotesLayer | null;
+    getLayer(layerName: 'TokenLayer'): TokenLayer | null;
+    getLayer(layerName: 'ForegroundLayer'): ForegroundLayer | null;
+    getLayer(layerName: 'SoundsLayer'): SoundsLayer | null;
+    getLayer(layerName: 'LightingLayer'): LightingLayer | null;
+    getLayer(layerName: 'SightLayer'): SightLayer | null;
+    getLayer(layerName: 'EffectsLayer'): EffectsLayer | null;
+    getLayer(layerName: 'ControlsLayer'): ControlsLayer | null;
+    getLayer(layerName: string): null;
```

Comment:
This makes it kind of not very extendable. Maybe we can improve that? at least it could return `CanvasLayer | null` or something like this.